src/main/java/dev/morling/onebrc/CalculateAverage_merykitty.java

```diff
@@ -0,0 +1,368 @@
+        }
+
+        Aggregator indexSimple(long offset, long size, int hash) {
+            hash = rehash(hash);
+            int bucketMask = nodes.length - 1;
+            int bucket = hash & bucketMask;
+            for (;; bucket = (bucket + 1) & bucketMask) {
+                PoorManMapNode node = nodes[bucket];
+                if (node == null) {
+                    this.size++;
+                    if (this.size * R_LOAD_FACTOR > nodes.length) {
+                        grow();
+                        bucketMask = nodes.length - 1;
+                        for (bucket = hash & bucketMask; nodes[bucket] != null; bucket = (bucket + 1) & bucketMask) {
+                        }
+                    }
+                    node = new PoorManMapNode(this.data, offset, size, hash);
+                    nodes[bucket] = node;
+                    return node.aggr;
+                }
+                else if (keyEqualScalar(node, offset, size, hash)) {
+                    return node.aggr;
+                }
+            }
+        }
+
+        void grow() {
+            var oldNodes = this.nodes;
+            var newNodes = new PoorManMapNode[oldNodes.length * 2];
+            int bucketMask = newNodes.length - 1;
+            for (var node : oldNodes) {
+                if (node == null) {
+                    continue;
+                }
+                int bucket = node.hash & bucketMask;
+                for (; newNodes[bucket] != null; bucket = (bucket + 1) & bucketMask) {
+                }
+                newNodes[bucket] = node;
+            }
+            this.nodes = newNodes;
+        }
+
+        static int rehash(int x) {
+            x = ((x >>> 16) ^ x) * 0x45d9f3b;
+            x = ((x >>> 16) ^ x) * 0x45d9f3b;
+            x = (x >>> 16) ^ x;
+            return x;
+        }
+
+        private boolean keyEqualScalar(PoorManMapNode node, long offset, long size, int hash) {
+            if (node.hash != hash || node.size != size) {
+                return false;
+            }
+
+            // Be simple
+            for (int i = 0; i < size; i++) {
+                int c1 = node.data[BYTE_SPECIES.vectorByteSize() + i];
+                int c2 = data.get(ValueLayout.JAVA_BYTE, offset + i);
+                if (c1 != c2) {
+                    return false;
+                }
+            }
+            return true;
+        }
+    }
+
+    private static long parseDataPoint(Aggregator aggr, MemorySegment data, long offset) {
+        long word = data.get(JAVA_LONG_LT, offset);
+        // This can be 12, 20, 28
+        int decimalSepPos = Long.numberOfTrailingZeros(~word & 0x10101000);
+        int shift = 28 - decimalSepPos;
+        // signed is -1 if negative, 0 otherwise
+        long signed = (~word << 59) >> 63;
+        long designMask = ~(signed & 0xFF);
+        long digits = ((word & designMask) << shift) & 0x0F000F0F00L;
+
+        // Now digits is in the form 0xUU00TTHH00
```

Comment:
What is TT, HH, etc? As for the main loop, could you provide an overview of how this thing works?

Reply:
UU stands for the value of the units digit, TT is the tens digit and HH is the hundreds digit.